apps/api/docs/database-design.md

```diff
@@ -0,0 +1,52 @@
+# **Database Design**
+
+This document provides a comprehensive and structured overview of the database design for Transcript Summarization App.
+
+It serves as a reference guide for stakeholders, including developers, database administrators, project managers, and other parties involved in the development, maintenance, and understanding of the application's database.
+
+The tables discussed below will be created as part of the database migration.
+
+## **Database Schema**
+
+The database schema consists of the following 3 tables:
+
+- **Db_uploads :** Contains details about all the user uploaded files with their Job IDs.
+- **Db_queue_logs** : Contains details about the queue system-related data, such as job metadata and status.
+- **Db_monitoring_logs** : Contains details about all the monitoring-related data, such as logs, response status etc.
+
+This schema can be visualized in the following image:
+![Block Diagram](./assets/transcript-summarization-database-schema.jpg)
+
+## **Data Dictionary**
+
+**db_uploads**
+
+| Attribute | Data Type | Not Null | Default | Description |
+| --- | --- | --- | --- | --- |
+| job_id | int(11) | True |     | Primary key, unique identifier for the uploaded record |
+| file_path | varchar(255) |     |     | File path of summarized file stored in server after successful processing |
+| created_on | timestamp | True | current_timestamp() | Stores the timestamp for the creation of the record |
+| created_by | varchar(45) | True |     | Stores the details of creator of the record |
+| modified_on | timestamp | True | current_timestamp() | Stores the timestamp for the last update to the record |
+| modified_by | varchar(45) |     |     | Stores the details of modifier of the record |
+
+**db_queue_logs**
+
+| Attribute | Data Type | Not Null | Default | Description |
+| --- | --- | --- | --- | --- |
+| queue_id | int(11) | True |     | Primary key, unique identifier for the uploaded record |
+| job_id | varchar(255) | True |     | Stores the ID value for related job_id |
+| status | enum | True |     | Stores the response status code of the response |
+| created_on | timestamp | True | current_timestamp() | Stores the timestamp for the creation of the record |
+| modified_on | timestamp | True | current_timestamp() | Stores the timestamp for the last update to the record |
+
+**db_monitoring_logs**
+
+| Attribute | Data Type | Not Null | Default | Description |
```

Comment:
We should only have a jobs table and file  table
Jobs table will have details related to file
so a autogenerated PK
input fileId => Which will be foreign key linked with file.
output fileId => Foreign Key to 
job_status => Another Enum which will have values like PENDING(added record in DB), QUEUED(When pushed to Queue in Cron JOB), PROCESSING(When the consumer pulls the job from Queue to Process), FINISHED(When consumer finished the job and output file is available)

File table will have file related columsn like path, file name.

These will be common fields
status => This is 1 or 0 for ACTIVE and INACTIVE as we will be doing soft delete.
created_on =>  Time stamp value
modified_on => Time stamp value
created_by => Keep it string for now as "admin"
updated_by => Keep it string for now as "admin"

Also try creating the diagram in MySQL Workbench its easier to create and you get better diagram too